app/src/main/java/net/squanchy/navigation/RoutingActivity.java

```diff
@@ -0,0 +1,31 @@
+package net.squanchy.navigation;
+
+import android.content.Intent;
+
+import net.squanchy.fonts.TypefaceStyleableActivity;
+import net.squanchy.navigation.deeplink.DeepLinkRouter;
+
+import timber.log.Timber;
+
+public class RoutingActivity extends TypefaceStyleableActivity {
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        RoutingComponent component = RoutingInjector.obtain(this);
+        DeepLinkRouter deepLinkRouter = component.deepLinkRouter();
+        Navigator navigator = component.navigator();
+
+        Intent intent = getIntent();
+        if (deepLinkRouter.hasDeepLink(intent)) {
```

Comment:
Actually, no, it won't work, it would break out the encapsulation of the parsing of the intent and expose the intent being launched, which is basically the opposite to what it is right now. The idea would only work if the `DeepLinkRouter` was simply a `DeepLinkParser` and it left the job of creating the intents and launching them to the activity, which I'd rather not do.